docs/src/main/asciidoc/config-extending-support.adoc

```diff
@@ -0,0 +1,338 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Extending Configuration Support
+
+include::./attributes.adoc[]
+
+:numbered:
+:sectnums:
+:sectnumlevels: 4
+:toc:
+
+[[custom-config-source]]
+== Custom Config Sources
+
+It is possible to create a Custom Config Source as specified in
```

Comment:
"Custom Config Source" isn't a proper noun, so this use of capitalization is nonstandard.

It should be `CustomConfigSource` if this is a type name, or "custom configuration source".

Reply:
Also (minor) I would prefer "It's" here to "It is" which sounds stilted.